python/ray/data/dataset.py

```diff
@@ -2395,6 +2395,144 @@ def iter_batches(
 
         stats.iter_total_s.add(time.perf_counter() - time_start)
 
+    def iter_torch_batches(
+        self,
+        *,
+        prefetch_blocks: int = 0,
+        batch_size: Optional[int] = None,
+        batch_format: str = "native",
+        drop_last: bool = False,
+        local_shuffle_buffer_size: Optional[int] = None,
+        local_shuffle_seed: Optional[int] = None,
+    ) -> Iterator[Union["torch.Tensor", Dict[str, "torch.Tensor"]]]:
```

Comment:
Shall we define a type `TorchTensorBatchType` for `Union["torch.Tensor", Dict[str, "torch.Tensor"]]`?